{% extends "app/layout.html" %}

{% block head-scripts %}
{% if iframe == True %}
<style>
nav {
    display: none;
}
body {
    padding-top: 0px;
}
</style>
{% endif %}
{% endblock %}

{% block scripts %}
    {% load staticfiles %}
    <link type="text/css" rel="stylesheet" media="all" href="{% static 'blast/css/main.css' %}" />
    <script type="text/javascript" src="{% static 'blast/scripts/jquery-1.11.1.min.js' %}"></script>
    <script type="text/javascript" src="{% static 'blast/scripts/jquery.hoverIntent.minified.js' %}"></script>
    <script>
    var dataset_list = {{ blastdb_list|safe }};
    </script>
    <script type="text/javascript" src="{% static 'blast/scripts/blast-multi.js' %}"></script>
    <script type="text/javascript" src="{% static 'blast/scripts/jquery.validate.min.js' %}"></script>
    
{% endblock %}

{% block content %}
<div class="tripal_details">
<FORM METHOD="POST" id="MainBlastForm" NAME="MainBlastForm" ENCTYPE="multipart/form-data">
{% csrf_token %}
<div class="tripal_organism-info-box tripal-info-box">
<!--<a href="docs/blast_databases.html">-->
<fieldset id="fieldset-datasets">
	<legend>BLAST Databases</legend>
	<fieldset class="organism" id="fieldset-organism">
		<legend id="legend-Organisms">Organisms</legend>
		<div id="box-organism">
			<label><div organism="all" class="all organism-div organism-active-background"><input type="checkbox" organism="all" class="all-organism-checkbox">All organisms</div></label>
		</div>
	</fieldset>
	<div id="box-datasets">
		<div class="datasets-div all">
			<div class="dataset-title">All organisms</div>
			<fieldset class="nucleotide">
				<legend>Nucleotide</legend>
				<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="genome-assembly" class="all-dataset-checkbox nucleotide genome-assembly">Genome Assembly - {{ blastdb_type_counts.genome_assembly }} databases</label></div>
				<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="transcript" class="all-dataset-checkbox nucleotide transcript">Transcript - {{ blastdb_type_counts.transcript }} databases</label></div>
			</fieldset>
			<fieldset class="peptide">
				<legend>Peptide</legend>
				<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="protein" class="all-dataset-checkbox peptide protein">Protein - {{ blastdb_type_counts.protein }} databases</label></div>
			</fieldset>
		</div>
	</div>
</fieldset>

<!--
Enter here your input data as 
<select name = "INPUT_TYPE"> 
    <option> Sequence in FASTA format 
    <option> Accession or GI 
</select>
-->

<fieldset id="fieldset-query">
	<legend>Query Sequence</legend>
	<div class="enter-query-text">Enter sequence below in <a target="_blank" href="http://en.wikipedia.org/wiki/FASTA_format">FASTA</a> format:</div>
	<textarea id="query-textarea" name="query-sequence" rows=6 cols=60></textarea>
	<BR>
	<span class="load-nucleotide">Or</span> <span class="load-peptide">load</span> it from disk 
	<input type="file" name="query-file" class="query-file">
    <!--
	<div>
		Set subsequence: From <input TYPE="text" NAME="QUERY_FROM" VALUE="" SIZE="10"> To <input TYPE="text" NAME="QUERY_TO" VALUE="" SIZE="10">
	</div>
    -->
</fieldset>

<fieldset id="fieldset-program">
	<legend>Program</legend>
	<label class="program blastn">
		<input type="radio" name="program" value="blastn" checked="checked" class="program blastn">
		blastn
	</label>
	<label class="program tblastn">
		<input type="radio" name="program" value="tblastn" class="program tblastn">
		tblastn
	</label>
	<label class="program tblastx">
		<input type="radio" name="program" value="tblastx" class="program tblastx">
		tblastx
	</label>
	<label class="program blastp">
		<input type="radio" name="program" value="blastp" class="program blastp">
		blastp
	</label>
	<label class="program blastx">
		<input type="radio" name="program" value="blastx" class="program blastx">
		blastx
	</label>
	<div id="blastProgramDescription">blastn - Nucleotide vs. Nucleotide</div>
	<div id="buttons">
		<INPUT TYPE="button" VALUE="Reset" CLASS="btn_reset">
		<INPUT TYPE="button" VALUE="Search" onclick="On_Submit();">
	</div>
</fieldset>

<!-- BLAST options start -->
<div id="blast_options">
    <fieldset id="fieldset-options-blast">
        <legend></legend>
        <fieldset id="fieldset-options-blast-general-param">        
        <legend>General Parameters</legend>    
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input TYPE="text" ID="evalue" NAME="evalue" VALUE="10.0" SIZE="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" ID="span_word_size" title="Length of initial exact match. (Default: 28)">Word size </span>
          </td>
          <td>
            <input TYPE="text" ID="word_size" NAME="word_size" VALUE="28" SIZE="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>
        
        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
        </fieldset>
        
        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input TYPE="checkbox" ID="low_complexity" VALUE="yes" NAME="low_complexity"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences."> Low complexity</span><br/>
        <input TYPE="checkbox" ID="soft_masking" VALUE="true" NAME="soft_masking"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches)"> Mask for lookup table only</span><br/>
        </fieldset>
    </fieldset>
</div>
<!-- BLAST options end -->
   
	<div>
		<INPUT TYPE="button" VALUE="Reset" CLASS="btn_reset">
		<INPUT TYPE="button" VALUE="Search" onclick="On_Submit();">
	</div>

</FORM>
</div>
</div>
{% endblock %}
